src/liquid/components/ld-select/ld-select.tsx

```diff
@@ -363,17 +366,25 @@ export class LdSelect {
     }
 
     if (!initialized) {
-      this.internalOptionsHTML = this.el.innerHTML['replaceAll'](
-        /<ld-option/g,
-        `<ld-option-internal${this.multiple ? ' mode="checkbox"' : ''}${
+      let internalOptionsHTML = ''
+      for (const ldOption of children) {
+        const classStr = ldOption.classList.length
+          ? ldOption.classList.toString()
+          : ''
```

Comment:
```suggestion
        const classStr = ldOption.classList.toString()
```

Empty classList results in `''`.